Model/User.hs

```diff
@@ -44,9 +45,11 @@ upsertUser user = entityKey <$> upsert user
     , UserEmail =. userEmail user
     ]
 
-credsToUser :: Creds m -> Maybe User
-credsToUser Creds{..} = User
+credsToUser :: Maybe PlanId -> Creds m -> Maybe User
+credsToUser mplanId Creds{..} = User
     <$> lookup "name" credsExtra
     <*> lookup "email" credsExtra
     <*> pure credsPlugin
     <*> pure credsIdent
+    <*> mplanId
+    <*> pure Nothing
```

Comment:
Nope, totally unrelated.

https://www.haskell.org/hoogle/?hoogle=pure

`pure` is from `Applicative`. It puts a value in a default, minimal context: `Applicative f => a -> f a`. In this case, the `f` is `Maybe` and it's implementation of `pure` is `Just`. Since the type of that expression needs to be `Maybe (Maybe S.CustomerId)`, `pure Nothing` works.

https://www.haskell.org/hoogle/?hoogle=liftIO

`liftIO` on the other hand takes an `IO` action and allows you to perform it in any monad that has a `MonadIO` instance: `liftIO :: MonadIO m => IO a -> m a`. It's common to do that in `Yesod` apps because you may want to perform `IO` from within `Handler` or `Widget`. Both have `MonadIO` instances, so `liftIO` works.